src/map/unit.cpp

```diff
@@ -1831,16 +1833,35 @@ void unit_set_attackdelay(block_list& bl, t_tick tick)
 
 	switch (bl.type) {
 		case BL_PC:
-			ud->attackabletime = tick + status_get_adelay(&bl);
-			// A fixed delay is added here which is equal to the minimum attack motion you can get
-			// This ensures that at max ASPD attackabletime and canact_tick are equal
-			ud->canact_tick = tick + status_get_amotion(&bl) + (pc_maxaspd(reinterpret_cast<map_session_data*>(&bl)) / AMOTION_DIVIDER_PC);
+			switch (event) {
+				case DELAY_EVENT_ATTACK:
+				//case DELAY_EVENT_CASTBEGIN_ID:
+				//case DELAY_EVENT_CASTBEGIN_POS:
+				case DELAY_EVENT_PARRY:
+					ud->attackabletime = tick + status_get_adelay(&bl);
+					// A fixed delay is added here which is equal to the minimum attack motion you can get
+					// This ensures that at max ASPD attackabletime and canact_tick are equal
+					ud->canact_tick = tick + status_get_amotion(&bl) + (pc_maxaspd(reinterpret_cast<map_session_data*>(&bl)) / AMOTION_DIVIDER_PC);
+					break;
+			}
 			break;
 		case BL_MER:
-			// TODO: Should set this, but only for ground skills
```

Comment:
Could be restored?

Reply:
I moved the code solution to the corresponding issue on github instead.